packages/webapp/src/components/Task/AddSoilAmendmentProducts/QuantityApplicationRate/index.tsx

```diff
@@ -0,0 +1,238 @@
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowUp';
+import { ReactComponent as SwapIcon } from '../../../../assets/images/swap.svg';
+import TextButton from '../../../Form/Button/TextButton';
+import Switch from '../../../Form/Switch';
+import Unit from '../../../Form/Unit';
+import NumberInput from '../../../Form/NumberInput';
+import {
+  location_area,
+  soilAmounts,
+  soilAmountsVolume,
+  applicationRateWeight,
+  applicationRateVolume,
+} from '../../../../util/convert-units/unit';
+import { TASK_PRODUCT_FIELD_NAMES, type TaskProductFormFields } from '../types';
+import { AreaApplicationSummary } from './AreaApplicationSummary';
+import { useQuantityApplicationRate } from './useQuantityApplicationRate';
+
+const SOIL_AMENDMENT_UNITS = {
+  WEIGHT: {
+    units: soilAmounts, // original
+  },
+  VOLUME: {
+    units: soilAmountsVolume,
+  },
+  APPLICATION_RATE_WEIGHT: {
+    units: applicationRateWeight,
+  },
+  APPLICATION_RATE_VOLUME: {
+    units: applicationRateVolume,
+  },
+};
+
+export interface Location {
+  type: string;
+  total_area: number;
+  total_area_unit: 'm2' | 'ha' | 'ft2' | 'ac';
+}
+
+export type QuantityApplicationRateProps = {
+  productId: number | string;
+  isReadOnly: boolean;
+  system: 'metric' | 'imperial';
+  location: Location;
+  defaultValues?: TaskProductFormFields;
+};
+
+const QuantityApplicationRate = ({
+  productId,
+  isReadOnly,
+  system, // measurementSelector
+  location,
+  defaultValues,
+}: QuantityApplicationRateProps) => {
+  const { t } = useTranslation();
+
+  const [isExpanded, setIsExpanded] = useState(false);
+  const toggleExpanded = () => setIsExpanded((prev) => !prev);
+
+  const { total_area, total_area_unit, type } = location;
+
+  const { control, getValues, setValue, register, watch } = useForm<TaskProductFormFields>({
+    defaultValues: {
+      [TASK_PRODUCT_FIELD_NAMES.PERCENT_OF_LOCATION_AMENDED]: 100,
+      ...defaultValues,
+    },
+  });
+
+  /* Control of relationship between quantity, area, and application rate */
+  const {
+    onQuantityChange,
+    onApplicationRateChange,
+    onPercentLocationChange,
+    isWeight,
+    toggleMeasure,
+    previewStringValue,
+    previewStringUnit,
+  } = useQuantityApplicationRate({
+    total_area,
+    total_area_unit,
+    system,
+    watch,
+    setValue,
+    getValues,
+  });
+
+  const percent_of_location = watch(TASK_PRODUCT_FIELD_NAMES.PERCENT_OF_LOCATION_AMENDED);
+
+  /* Label for weight/volume switch */
+  const formatLabel = (labelKey: string, shouldBeBold: boolean) => {
+    // t('ADD_TASK.SOIL_AMENDMENT_VIEW.WEIGHT')
+    // t('ADD_TASK.SOIL_AMENDMENT_VIEW.VOLUME')
+    const label = t(labelKey);
+    return shouldBeBold ? (
+      <span className={styles.switchLabel}>
+        <b>{label}</b>
+      </span>
+    ) : (
+      <span className={styles.switchLabel}>{label}</span>
+    );
+  };
+
+  return (
+    <div className={styles.container}>
+      <Switch
```

Comment:
I think this is missing the disabled / readonly prop. I was expecting it to be disabled when the other inputs are as well.

Reply:
Oh yeah, it definitely should 🤦‍♀️